### PR TITLE
Full-stack group Capstone Project - FrontEnd

### PR DESCRIPTION
**Pull request by dfsalam**
Hello 👋.
Link to the backend PR: [Back-End PR Link](https://github.com/dennis-every/book-an-appointment-api/pull/20)
Our group members are @ashmalzahra, @Nkaleth, @dennis-every and @dfsalam, we implemented those features:
Core features - Required for every team size
1. The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
2. In the navigation panel, the user can see links to:
"Places"
"Reserve" form.
"My reservations".
"Add places"
"Delete places"
3. On the main page, the user can see a list of places.
4. When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description.
In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

5. When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

6. Make the app responsive, creating both mobile and desktop versions.

7. When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
- When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city and add all necessary inputs.
- The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.
Booking

8. When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).
9. Add full documentation for your API.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import DeletePlace from './routes/DeletePlace';
+import MyReservations from './routes/MyReservations';
+import Reserve from './routes/Reserve';
+import Places from './routes/Places';
+import NotMatch from './routes/NotMatch';
+import Layout from './components/Layout';
+import LogIn from './routes/LogIn';
+import PlaceDetails from './routes/PlaceDetails';
+
+function App() {
```
Comment (Meltrust) [inline src/App.jsx]: - [ ]  Please, kindly refactor this main `App` component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

File: README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+
```
Comment (Meltrust) [inline README.md:1]: - [x] Kindly, include in your instructions the name of your test user "John", so your users can test your app without having to create users in the Rails console. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @dfsalam,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful, gorgeous, nice frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/NavigationBar.jsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                      {' '}
+                      LOG OUT
+                    </Nav.Link>
+                  </>
+                ) : (
+                  <Nav.Link className="link flex capital-text" as={NavLink} to="/login" onClick={toggleSidebar}>
+                    <i className="bx bx-sm bx-log-in" />
+                    {' '}
+                    LOG IN
+                  </Nav.Link>
```
Comment (mcrd25) [inline src/components/NavigationBar.jsx]: [OPTIONAL]

There may be unnecessary repetition here. These links seem to be similar, maybe the code can be condensed with some sort of looping, and array

OR abstraction through a smaller component like so:


```js
const SidebarLink = ({ to, icon, label, onClick }) => (
  <li className="item">
    <Nav.Link className="link flex capital-text" as={NavLink} to={to} onClick={onClick}>
      <i className={`bx bx-sm ${icon}`} />
      {' '}
      {label}
    </Nav.Link>
  </li>
);

.....

  return (
    // ... remains the same, replace Nav.Link items with SidebarLink
    <SidebarLink to="/" icon="bxs-building-house" label="PLACES" onClick={toggleSidebar} />
    // ...
  );

```

or BOTH.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Great job with requested changes from the previous reviewer 👍 
- And even though it's been said before, your UI is pretty neat!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- Consider adding a sign up page to the front-end (username only), since a regular user would not have access to the back-end to create a user through the api [Please address this change]
- [OPTIONAL] Seeing where you can make code better by following best practices: See the following articles
   - https://www.freecodecamp.org/news/best-practices-for-react/
   - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb
- [OPTIONAL] There are some candidates to make some of these html components reusable.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — APPROVED** [review]
#### Hi @dfsalam,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme has been improved✔️ 
- Previous reviewer's suggestion about repetition has been taken into action✔️ 
- The main `App` component is now ES6✔️
- Great work ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**